tests/api_test.py

```diff
@@ -4882,6 +4882,28 @@ def f_jvp(primal, tangent):
 
     api.vmap(sample)(jax.random.split(jax.random.PRNGKey(1), 3))  # don't crash
 
+  def test_closure_with_vmap2(self):
+    # https://github.com/google/jax/issues/8783
+    def h(z):
+      def f(x):
+        @jax.custom_jvp
+        def g(y):
+          return x * y
```

Comment:
Maybe a test that returns only `x` here? Since the implementation of `*` might involve call primitives for transformations (namely `xla_call` for `jit`).